resim/msg/converter_plugin.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include "resim/msg/converter_plugin.hh"
+
+#include <dlfcn.h>
+
+#include <utility>
+
+#include "resim/assert/assert.hh"
+
+namespace resim::msg {
+
+ConverterPlugin::ConverterPlugin(const std::filesystem::path &plugin_path)
+    : handle_(dlopen(plugin_path.c_str(), RTLD_LAZY)) {
+  REASSERT(handle_ != nullptr, "Failed to load plugin!");
+
+  converter_ =
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+      reinterpret_cast<const Converter>(dlsym(handle_, CONVERTER_NAME));
+  REASSERT(converter_ != nullptr, "Failed to load_plugin!");
+}
+
+ConverterPlugin::~ConverterPlugin() { dlclose(handle_); }
```

Comment:
```suggestion
ConverterPlugin::~ConverterPlugin() { 
  if (handle_) {
    dlclose(handle_);
  } 
}
```

Reply:
This was originally a codecov issue that I decided to avoid, but I added a friend test to get coverage of it and put the branch back.